config/build.ts

```diff
@@ -91,10 +77,11 @@ async function processSnippets(dir, processedSnippets) {
 
     // Helpers:
 
-    async function processAndValidateSnippet(file: SnippetFileInput, dir: string): Promise<SnippetProcessedData> {
+    async function processAndValidateSnippet(file: SnippetFileInput): Promise<SnippetProcessedData> {
         const messages: Array<string | Error> = [];
         try {
             status.add(`Processing ${file.relativePath}`);
+            let dir = file.type === 'private' ? 'private-samples' : 'samples';
```

Comment:
This feels like some weird reverse logic.  Shouldn't the file know its dir, rather than you deriving the path based on the file type.